### DISCUSSION TITLE
InvalidOperationException: Generated PDF file is empty in Serenity 6.9.6

### DISCUSSION THREAD

**Discussion by flashyjunior** [serenity-is/Serenity#7115] [Help]
Generating pdf using wkhtmltopdf.exe works fine on development environment, after deploying to a cloud server it works fine if you run the application directly from the cloud server, but I get this error message whenever i access the site outside the server that hosts the application.
Please help...

![image](https://github.com/serenity-is/Serenity/assets/7760804/32932858-1ef0-4478-846f-8c32b3a547d0)

**Comment by flashyjunior** [comment]
Ok, so i found out that am accessing the Virtual private Server via public IP, maybe that could be the issue, I deployed to a shared hosting and the report generates successfully.
How can I specify the local IP so the `wkhtmltopdf `can generate the PDF

**Reply by LSNicholls** [reply]
Maybe the proxy information in the docs would help? https://wkhtmltopdf.org/usage/wkhtmltopdf.txt Just a guess
